File: scripts/binarySearch.js
const binarySearch = (list, item) => {
    let low = 0;
    let high = list.length - 1

    while(low <= high) {
        const middle = (low + high) / 2
        const guess = list[middle]
        
        if(guess === item) {
            return middle
        } else if(guess < item) {
            low = middle + 1
        } else {
            high = middle - 1
        }
    }
    return none
}

const list = [1, 2, 3, 4, 5, 6, 7]
const item = 5

console.log(binarySearch(list, item))

File: python-algorithms/binary_search.py
def binary_search(arr, item):
    low = 0
    high = len(arr) - 1
    
    while low <= high:
        middle = round((low + high) / 2)
        guess = arr[middle]
        
        if guess == item:
            return middle
        if guess > item:
            high = middle - 1
        else:
            low = middle + 1
    return None
    
my_arr = [1, 3, 4, 6, 7, 20]

print(binary_search(my_arr, 3))
